source/MaterialXRuntime/Codegen/RtSourceCodeImpl.h

```diff
@@ -0,0 +1,60 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved. See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_RTSOURCECODEIMPL_H
+#define MATERIALX_RTSOURCECODEIMPL_H
+
+/// @file RtSourceCodeImpl.h
+/// TODO: Docs
+
+#include <MaterialXRuntime/Codegen/RtCodegenImpl.h>
+
+namespace MaterialX
+{
+
+/// @class RtSourceCodeImpl
+/// Node implementation using shader source code.
+class RtSourceCodeImpl : public RtCodegenImpl
+{
+    DECLARE_TYPED_SCHEMA(RtSourceCodeImpl)
+
+public:
+    /// Constructor.
+    RtSourceCodeImpl(const RtPrim& prim) : RtCodegenImpl(prim) {}
+
+    /// Set a string containing the source code to use for this implementation.
+    void setFile(const string& file);
+
+    /// Return a string containing the source code used by this implementation.
+    const string& getFile() const;
+
+    /// Set a string containing the source code to use for this implementation.
```

Comment:
Don't you have the same comments for set/get file as set/get sourceCode? Shouldn't they be different?

Reply:
Good catch, I will fix!